assignments/lambda-classes.js

```diff
@@ -1 +1,135 @@
+console.log(instructor1);
+instructor1.demo('Classes');
+instructor1.grade({ name: 'David' }, 'Javascript');
+
+const instructor2 = new Instructor({ name: 'Sarah', age: 22, location: 'Boston', gender: 'Female', specialty: 'Algorithms', favLanguage: 'Python', catchPhrase: 'This is going to be fun' });
+console.log(instructor2);
+instructor2.demo('Python');
+instructor2.grade({ name: 'Juan' }, 'Python');
+
+
+class Student extends Person {
+  constructor(object) {
+    super(object);
+    this.previousBackground = object.previousBackground;
+    this.className = object.className;
+    this.favSubjects = object.favSubjects;
+    this.grade = undefined; // Stretch problem
+  }
+
+  listsSubjects() {
+    this.favSubjects.map(subject => console.log(subject));
+  }
+
+  PRAssignment(subject) {
+    console.log(`${this.name} has submitted a PR for ${subject}`);
+  }
+
+  sprintChallenge(subject) {
+    console.log(`${this.name} has begun sprint challenge on ${subject}`);
+  }
+
+  graduate() {
+    if (this.grade >= 70) {
+      console.log(`Congratulations to ${this.name} for graduating!`);
+    } else {
+      console.log(`Oops, ${this.name} did not graduate`);
+      this.grade = undefined;
+    }
+  }
+}
+
+const student1 = new Student({ name: 'John', age: 32, location: 'California', gender: 'Male', previousBackground: 'Finance', className: 'WEBEU2', favSubjects: ['React', 'Data structures'] });
+console.log(student1);
+student1.listsSubjects();
+
+const student2 = new Student({ name: 'Sarah', age: 22, location: 'Boston', gender: 'Female', previousBackground: 'Designer', className: 'WEBEU2', favSubjects: ['CSS', 'Node JS'] });
+console.log(person2);
+student2.PRAssignment('CSS');
+student2.sprintChallenge('Advanced Javascript');
+
+
+class ProjectManager extends Instructor {
+  constructor(object) {
+    super(object);
+    this.gradClassName = object.gradClassName;
+    this.favInstructor = object.favInstructor;
+  }
+
+  standUp(channel) {
+    console.log(`${this.name} announces to ${channel}, @channel standy times!​​​​​`);
+  }
+
+  debugsCode(student, subject) {
+    console.log(`${this.name} debugs ${student.name}'s code on ${subject}`);
+  }
+}
+
+const pm1 = new ProjectManager({ name: 'Mike', age: 35, location: 'New York', gender: 'Male', specialty: 'HTML', favLanguage: 'Javascript', catchPhrase: 'Here we go!', gradClassName: 'WEBEU2', favInstructor: 'Gabe' });
+console.log(pm1);
+pm1.standUp('Webeu2_Mike');
+
+const pm2 = new ProjectManager({ name: 'Sarah', age: 22, location: 'Boston', gender: 'Female', specialty: 'Algorithms', favLanguage: 'Python', catchPhrase: 'This is going to be fun', gradClassName: 'WEBEU2', favInstructor: 'Gabe' });
+console.log(pm2);
+pm2.debugsCode(student1, 'Python');
+
+
+// Stretch problem
+while (student1.grade === undefined) {
+  instructor1.assignGrade(student1);
+  student1.graduate();
```

Comment:
Well done with stretch goals. They look great. 
It's really great to watch in the console the students try to graduate. I love it. 

I have to say tough I don't really like the choice to have these 2 while statements here in the wild. 
First thing that comes in my mind here would be to write a function that takes a student and an instructor and use your while  in the body. That should even resolve the fact that you have to declare a different statement every time you want a student to graduate.